server/src/main/java/org/opensearch/search/fetch/FetchPhase.java

```diff
@@ -103,101 +106,167 @@ public FetchPhase(List<FetchSubPhase> fetchSubPhases) {
     }
 
     public void execute(SearchContext context) {
-        if (LOGGER.isTraceEnabled()) {
-            LOGGER.trace("{}", new SearchContextSourcePrinter(context));
+        FetchProfileBreakdown breakdown = null;
+        if (context.getProfilers() != null) {
+            breakdown = context.getProfilers().getFetchProfiler().getQueryBreakdown("fetch");
         }
+        try {
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace("{}", new SearchContextSourcePrinter(context));
+            }
 
-        if (context.isCancelled()) {
-            throw new TaskCancelledException("cancelled task with reason: " + context.getTask().getReasonCancelled());
-        }
+            if (context.isCancelled()) {
+                throw new TaskCancelledException("cancelled task with reason: " + context.getTask().getReasonCancelled());
+            }
 
-        if (context.docIdsToLoadSize() == 0) {
-            // no individual hits to process, so we shortcut
-            context.fetchResult()
-                .hits(new SearchHits(new SearchHit[0], context.queryResult().getTotalHits(), context.queryResult().getMaxScore()));
-            return;
-        }
+            if (context.docIdsToLoadSize() == 0) {
+                // no individual hits to process, so we shortcut
+                context.fetchResult()
+                    .hits(new SearchHits(new SearchHit[0], context.queryResult().getTotalHits(), context.queryResult().getMaxScore()));
+                return;
+            }
 
-        DocIdToIndex[] docs = new DocIdToIndex[context.docIdsToLoadSize()];
-        for (int index = 0; index < context.docIdsToLoadSize(); index++) {
-            docs[index] = new DocIdToIndex(context.docIdsToLoad()[context.docIdsToLoadFrom() + index], index);
-        }
-        // make sure that we iterate in doc id order
-        Arrays.sort(docs);
+            DocIdToIndex[] docs = new DocIdToIndex[context.docIdsToLoadSize()];
+            for (int index = 0; index < context.docIdsToLoadSize(); index++) {
+                docs[index] = new DocIdToIndex(context.docIdsToLoad()[context.docIdsToLoadFrom() + index], index);
+            }
+            Arrays.sort(docs);
+
+            Map<String, Set<String>> storedToRequestedFields = new HashMap<>();
+            FieldsVisitor fieldsVisitor;
+            if (breakdown != null) {
+                Timer t = breakdown.getTimer(FetchTimingType.CREATE_STORED_FIELDS_VISITOR);
+                t.start();
+                fieldsVisitor = createStoredFieldsVisitor(context, storedToRequestedFields);
+                t.stop();
+            } else {
+                fieldsVisitor = createStoredFieldsVisitor(context, storedToRequestedFields);
+            }
```

Comment:
All of this if else block trying to profile, please refactor into one method.
Making it less cumbersome in this already huge method

All these if else + try finally seem mess up the diff view, making it really hard to see what are changed